Framework/PythonInterface/plugins/algorithms/CropWorkspaceRagged.py

```diff
@@ -0,0 +1,122 @@
+#pylint: disable=no-init,invalid-name
+from __future__ import (absolute_import, division, print_function)
+from mantid.api import *
+from mantid.simpleapi import ConjoinWorkspaces, CropWorkspace, RenameWorkspace
+from mantid.kernel import *
+import numpy as np
+
+
+class CropWorkspaceRagged(PythonAlgorithm):
+
+    def category(self):
+        return 'Transforms\\Splitting;Workflow'
+
+    def name(self):
+        return 'CropWorkspaceRagged'
+
+    def summary(self):
+        return 'Crop each spectrum of a workspace independently'
+
+    def PyInit(self):
+        self.declareProperty(WorkspaceProperty('InputWorkspace', '',
+                                               direction=Direction.Input), 'input workspace')
+        self.declareProperty(WorkspaceProperty('OutputWorkspace', '',
+                                               direction=Direction.Output), 'output workspace')
+        self.declareProperty(FloatArrayProperty('Xmin'), 'minimum x values with NaN meaning no minimum')
+        self.declareProperty(FloatArrayProperty('Xmax'), 'maximum x values with NaN meaning no maximum')
+
+    def validateInputs(self):
+        inputWS = self.getProperty('InputWorkspace').valueAsStr
+        xmins = self.getProperty('XMin').value
+        xmaxs = self.getProperty('XMax').value
+
+        numSpec = mtd[inputWS].getNumberHistograms()
+        numMin = len(xmins)
+        numMax = len(xmaxs)
+
+        errors = {}
+
+        if numMin == 0 and numMax == 0:
+            errors['XMin'] = 'Must specify minimums or maximums'
+            errors['XMax'] = 'Must specify minimums or maximums'
+
+        if numMin > 1 and numMin != numSpec:
+            errors['XMin'] = 'Must specify min for each spectra ({}!={})'.format(numMin, numSpec)
+
+        if numMax > 1 and numMax != numSpec:
+            errors['XMax'] = 'Must specify max for each spectra ({}!={})'.format(numMax, numSpec)
+
+        return errors
+
+    def PyExec(self):
+        inputWS = self.getProperty('InputWorkspace').value
+        outputWS = self.getProperty('OutputWorkspace').valueAsStr
+        xmins = self.getProperty('XMin').value
+        xmaxs = self.getProperty('XMax').value
+
+        if len(xmins) == 1 and len(xmaxs) == 1:
+            CropWorkspace(InputWorkspace=inputWS, OutputWorkspace=outputWS,
```

Comment:
With this line, when running the algorithm with `StoreInADS=False` the output workspace will still show up in the ADS. One solution would be to replace this line by
```python
name = '__{}_cropped_'.format(outputWS)
CropWorkspace(InputWorkspace=inputWS, OutputWorkspace=name, XMin=xmins[0], XMax=xmaxs[0])
outputWS = mtd[name]
mtd.remove(name)
```

and at the end of PyExec():
```python
self.setProperty('OutputWorkspace', outputWS)
```